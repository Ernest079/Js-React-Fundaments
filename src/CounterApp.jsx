import {useState} from 'react';


export const CounterApp = ({value}) => {
  if(!value){
    value = 0;
  }

  const [counter, setCounter] = useState(value);

  const buttonAdd = () => {
    // console.log(event);
    setCounter(counter + 1);  
  }
  const buttonLess = () => {
    // console.log(event);
    setCounter(counter - 1);  
  }
  const buttonReset = () => {
    // console.log(event);
    setCounter(value);  
  }

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>
      <button onClick={buttonAdd}>
        +1
      </button>
      <button onClick={buttonLess}>
        -1
      </button>
      <button aria-label='btn-reset' onClick={buttonReset}>
        Reset
      </button>
    </>
  );
}
