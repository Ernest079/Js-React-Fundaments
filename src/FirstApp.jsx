//Los proptypes estan obsoletos para la versiÃ³n 19 de react
import PropTypes from 'prop-types';

export const FirstApp = ({title = 'test', subTitle = 'test', name = 'Ernesto'}) => {

  return (
    <>
      <h1>{title}</h1>
      {/* <h2>{nombre}</h2> */}
      <h2>{subTitle}</h2>
      <p>{name}</p>
    </> 
  )
}
