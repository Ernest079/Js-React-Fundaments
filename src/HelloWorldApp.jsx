// function App() {
//   return <h1>H O L A ~ M U N D O</h1>
// } 

// export default App;

export const HelloWorldApp = () => {
  return (
    <h1>H O L A ~ M U N D O</h1>
  )
}
  
