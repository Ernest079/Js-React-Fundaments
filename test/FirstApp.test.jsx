import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en FirstApp', () => {
  test('Debe de hacer match con el snapshot', () => {
    const title="O mueres como héroe o vives lo suficiente para convertirte en villano"; 
    const subTitle="El Caballero de la Noche";  
    const name="Henry Dent";
    const {container} = render(<FirstApp title = {title}/>);

    expect(container).toMatchSnapshot();
  });
  test('Debe de mostrar el titulo en un h1', () => {
    const title="O mueres como héroe o vives lo suficiente para convertirte en villano"; 
    const subTitle="El Caballero de la Noche";  
    const name="Henry Dent";
    const {container, getByText} = render(<FirstApp title = {title}/>);
    expect(getByText(title)).toBeTruthy();
    const h1 = container.querySelector('h1');
    expect(h1.innerHTML).toContain(title);
  });
});