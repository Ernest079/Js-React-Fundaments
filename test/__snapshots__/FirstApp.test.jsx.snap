// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    O mueres como h√©roe o vives lo suficiente para convertirte en villano
  </h1>
  <h2>
    test
  </h2>
  <p>
    Ernesto
  </p>
</div>
`;
